java/java.lsp.server/test/unit/src/org/netbeans/modules/java/lsp/server/ServerTest.java

```diff
@@ -0,0 +1,569 @@
+                          "    line = 4\n" +
+                          "    character = 4\n" +
+                          "  ]\n" +
+                          "  end = Position [\n" +
+                          "    line = 7\n" +
+                          "    character = 5\n" +
+                          "  ]\n" +
+                          "]:(Method:innerMethod:Range [\n" +
+                          "  start = Position [\n" +
+                          "    line = 5\n" +
+                          "    character = 8\n" +
+                          "  ]\n" +
+                          "  end = Position [\n" +
+                          "    line = 6\n" +
+                          "    character = 9\n" +
+                          "  ]\n" +
+                          "]:()), Field:field:Range [\n" +
+                          "  start = Position [\n" +
+                          "    line = 1\n" +
+                          "    character = 4\n" +
+                          "  ]\n" +
+                          "  end = Position [\n" +
+                          "    line = 1\n" +
+                          "    character = 22\n" +
+                          "  ]\n" +
+                          "]:(), Method:method:Range [\n" +
+                          "  start = Position [\n" +
+                          "    line = 2\n" +
+                          "    character = 4\n" +
+                          "  ]\n" +
+                          "  end = Position [\n" +
+                          "    line = 3\n" +
+                          "    character = 5\n" +
+                          "  ]\n" +
+                          "]:())";
+        assertEquals(expected, textualSymbols);
+    }
+
+    private String toString(DocumentSymbol sym) {
+        return sym.getKind().toString() + ":" +
+               sym.getName() + ":" +
+               sym.getRange() + ":" +
+               sym.getChildren()
+                  .stream()
+                  .map(this::toString)
+                  .collect(Collectors.joining(", ", "(", ")"));
+    }
+
+    @Test
+    public void testGoToDefinition() throws Exception {
+        File src = new File(getWorkDir(), "Test.java");
+        src.getParentFile().mkdirs();
+        String code = "public class Test {\n" +
+                      "    private int field;\n" +
+                      "    public void method(int ppp) {\n" +
+                      "        System.err.println(field);\n" +
+                      "        System.err.println(ppp);\n" +
+                      "    }\n" +
+                      "}\n";
+        try (Writer w = new FileWriter(src)) {
+            w.write(code);
+        }
+        FileUtil.refreshFor(getWorkDir());
+        ServerSocket srv = new ServerSocket(0);
+        new Thread(() -> {
+            try {
+                Socket server = srv.accept();
+                Server.run(server.getInputStream(), server.getOutputStream());
+            } catch (Exception ex) {
```

Comment:
Jackpot:
warning: Can be replaced with multicatch or several catch clauses catching specific exceptions